app/models/package_manager/multiple_sources_base/provider_map.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module PackageManager
+  class MultipleSourcesBase < Base
+    # Given a set of prioritized ProviderInfo objects, determine the most
+    # likely upstream package repository for a package. This uses the
+    # packages' versions' repository_sources field to
+    class ProviderMap
+      # @param prioritized_provider_infos [Array(ProviderInfo)]
+      def initialize(*prioritized_provider_infos)
```

Comment:
🎱 Is there a way to make it more obvious to the upstream that the order in which you pass in ProviderInfos is actually meaningful?

Reply:
🎱 What do you think about forcing the upstream to pass in an array named something like "prioritized_provider_infos"? That would let the initialize method take in additional methods (future expansion of the class) as well as force the upstream to figure out what "prioritized_provider_infos" means before it can be passed in

Reply:
Sure, I can do that.